app/models/manageiq/providers/autosde/storage_manager/storage_service.rb

```diff
@@ -46,16 +47,32 @@ def raw_delete_storage_service
   end
 
   def raw_update_storage_service(options = {})
-    capability_value_list = options.slice(*ext_management_system.capabilities.keys).values
-    capability_value_list.delete("-1")
+    # send resources only if edited
+    current_resources = storage_resources.ids
+    selected_resources = options["storage_resource_id"].to_a.pluck("value").map(&:to_i)
+    resources = if current_resources.sort != selected_resources.sort
+                  ext_management_system.storage_resources.find(options["storage_resource_id"].to_a.pluck("value")).pluck(:ems_ref)
+                end
+    # send capability_value_list only if edited
+    selected_capabilities_refs = options.slice(*ext_management_system.capabilities.keys).values
+    selected_capabilities_refs.delete("-1")  # get rid of the N/A option
+    current_capabilities = ext_management_system.capabilities.slice(*capabilities.keys)
+    current_capabilities_refs = []
+    capabilities.each do |key, value|
+      current_capabilities[key].each do |cap|
+        current_capabilities_refs.push(cap["uuid"]) if cap["value"] == value
+      end
+    end
+    capability_value_list = if current_capabilities_refs.sort != selected_capabilities_refs.sort
+                              selected_capabilities_refs
+                            end
```

Comment:
@OrGur1987 this is complex enough now to require a spec test demonstrating the various cases that you are handling here

Reply:
u mean a spec that tests the creation hash received by `ext_management_system.class::EmsRefreshWorkflow.create_job()` at the end of this method in the various cases? 

can I mock `EmsRefreshWorkflow` and check what it receives from `raw_update_storage_service()`?

Reply:
More importantly what `services_pk_put` is receiving based on the various scenarios on current capabilities vs selected capabilities

Reply:
how can I mock/check that?
we don't have too many examples in our provider repo spec dir
can u perhaps point me to a useful example from another provider?
thanks

Reply:
hi @agrare, 
@jeffibm merged the ui for this issue

Reply:
@agrare reminder
thanks

Reply:
@OrGur1987 you are looking for examples of spec tests?  You would return a `double("autosde_client")` (https://www.rubyguides.com/2018/10/rspec-mocks/) so you would `expect(ems).to receive(:autosde_client).and_return(autosde_client)`
Then you can `expect(autosde_client).to receive(:ServiceUpdate).with(expected_args).and_return(something)`

Reply:
I thought it might be easier and cleaner to export the preprocessing of the selected resources and capabilities to a helper method (say `prepare_update_args`), and then only test that in the spec, since there's no need to actually test the `client`.

something like:
```ruby
  let(:ems) do
    FactoryBot.create(
      # copied from some example, not sure it's needed here:
      :autosde_storage_manager,
      :with_autosde_credentials,
      :hostname     => Rails.application.secrets.autosde[:appliance_host],
      
      # dummy capabilities:
      :capabilities => {
        "cap1": [
          {
            "uuid": "111",
            "value": "True"
          },
          {
            "uuid": "222",
            "value": "False"
          }
        ],
        "cap2": [
          {
            "uuid": "333",
            "value": "True"
          },
          {
            "uuid": "444",
            "value": "False"
          }
        ],
      }
    )
    end

    let(:test_resource1) do
      FactoryBot.create(:storage_resource, :ext_management_system => ems)
    end

    let(:test_resource2) do
      FactoryBot.create(:storage_resource, :ext_management_system => ems)
    end

    let(:test_service) do
      FactoryBot.create(
        :storage_service,
        :ext_management_system => ems,
        :storage_resources     => [test_resource1, test_resource2],
        :capabilities          => {"cap1": "True", "cap2": "False"}
        )
    end

it "calls prepare update with the existing resources and capabilities" do
  test_service.prepare_update_args(options = {existing_resources_and_capabilities})
  ...
  ...
end
``` 

but the created `test_service` seems to be `ManageIQ::StorageService` and not `StorageManager::StorageService`, and I can't access a `StorageManager::StorageService` method without adding it to the super as well..

is there a way to `FactoryBot.create` the `StorageManager::StorageService`?

Reply:
found out that specifying the type in the create args produces the desired child-model.
is the added spec ok with u?
I did find a tiny bug in the ui and will open another pr for it